packages/emitter-framework/src/typescript/components/value-expression.tsx

```diff
@@ -0,0 +1,54 @@
+import { Children } from "@alloy-js/core";
+import * as ts from "@alloy-js/typescript";
+import { Value } from "@typespec/compiler";
+
+/**
+ * Properties for the {@link ValueExpression} component.
+ */
+interface ValueExpressionProps {
+  /**
+   * The TypeSpec value to be converted to a JavaScript expression.
+   */
+  value: Value;
+}
+
+/**
+ * Generates a JavaScript value expression from a TypeSpec value.
+ * @param props properties for the value expression
+ * @returns {@link Children} representing the JavaScript value expression
+ */
+export function ValueExpression(props: ValueExpressionProps): Children {
+  switch (props.value.valueKind) {
+    case "StringValue":
+    case "BooleanValue":
+    case "NullValue":
+      return <ts.ValueExpression jsValue={props.value.value} />;
+    case "NumericValue":
+      if (props.value.value.asNumber()) {
+        return <ts.ValueExpression jsValue={props.value.value.asNumber()} />;
+      }
+      return <ts.ValueExpression jsValue={props.value.value.asBigInt()} />;
```

Comment:
might want to check `isInteger` and compilerassert otherwise like below

Reply:
Can you give me an example of valid TypeSpec where this could happen? Just want to make sure I'm not adding defensive code for a state that's not representable and it looks like `Numeric::parse` does some validation and throws. 

(although maybe there are other ways a program can get in this state that I'm not aware of of course)

Reply:
```ts
 it("large decimal (> Number.MAX_VALUE)", () => {
      expectNumericData(
        "123456789123456789.112233445566778899",
        123456789123456789112233445566778899n,
        18,
      );
    });
```

```tsp
const a = 123456789123456789.112233445566778899;
```

this for example in js would loose precision but numeric should have the exact value